### PR TITLE
[MM-57724] Convert `./components/channel_header_dropdown/mobile_channel_header_dropdown.tsx` from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Syed-Ali-Abbas-Zaidi**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/26692
Jira https://mattermost.atlassian.net/browse/MM-57724

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/channel_header_dropdown/mobile_channel_header_dropdown.tsx [hunk 1]
```diff
@@ -46,39 +52,36 @@ class MobileChannelHeaderDropdown extends React.PureComponent<Props> {
...
+                <ChannelHeaderDropdownItems isMobile={true}/>
+                <div className='Menu__close visible-xs-block'>
+                    {'×'}
+                </div>
+            </Menu>
+        </MenuWrapper>
+    );
+};
 
 export default injectIntl(MobileChannelHeaderDropdown);
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/mobile_channel_header_dropdown.tsx:86]: Let's remove the inject and use `useIntl` instead.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just one minor thing.

**Review by devinbinnie — APPROVED** [review]
LGTM other than @larkox's comment. Thanks @Syed-Ali-Abbas-Zaidi!

**Review by larkox — APPROVED** [review]
Looks great now!

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**